extensions/lifecycle/tasks/LifecycleUpdateExpirationTask.js

```diff
@@ -152,8 +152,15 @@ class LifecycleUpdateExpirationTask extends BackbeatTask {
                     Object.keys(locations).find(name => locations[name].isCold);
 
                 const archive = objMD.getArchive();
-                // Reset archive flags to no longer
-                // show it as restored
+
+                // Confirm the object has indeed expired: it can happen that the
+                // expiration date is updated while the expiry was "in-flight" (e.g.
+                // queued for expiry but not yet expired)
+                if (new Date(archive.getRestoreWillExpireAt()) > new Date(Date.now())) {
```

Comment:
```suggestion
                if (new Date(archive.getRestoreWillExpireAt()) > new Date()) {
```
`new Date()` already returns the current date no need to add the `Date.now()`